mlflow/utils/file_utils.py

```diff
@@ -526,6 +529,9 @@ def get_or_create_nfs_tmp_dir():
         os.makedirs(tmp_nfs_dir, exist_ok=True)
     else:
         tmp_nfs_dir = tempfile.mkdtemp(dir=nfs_root_dir)
+        # mkdtemp creates a directory with permission 0o700
+        # change it to be 0o777 to ensure it can be seen in spark UDF
```

Comment:
Does the temporary directory have to be writable?

Reply:
Because "shutils" makedirs also use "0o777", I keep it the same with it

Reply:
So it has to be writable?

Reply:
I don't think it strictly needs to be writeable, but this was the preexisting behavior for the Spark model cache. To avoid regressions, I think it's best to use the same permissions level.

Reply:
Makes sense! I asked this because `can be seen` sounds like `should be readable but doesn't have to be writable`.